modules/ducktests/tests/ignitetest/tests/snapshot_test.py

```diff
@@ -0,0 +1,221 @@
+
+    SNAPSHOT_NAME = "test_snap"
+
+    @cluster(num_nodes=NUM_NODES)
+    @ignite_versions(str(DEV_BRANCH))
+    def snapshot_test(self, ignite_version):
+        """
+        Test Snapshot.
+        1. Start of ignite cluster.
+        2. Activate cluster.
+        3. Load.
+        4. Idle verify dump1.
+        5. Snapshot.
+        6. Load.
+        7. Idle verify dump2.
+        8. Сhecking that dump1 and dump2 are different.
+        9. Stop ignite and replace db from snapshot.
+        10. Run ignite cluster.
+        11. Idle verify dump3.
+        12. Checking the equality of dump1 and dump3.
+        """
+        data_cfg = DataStorageConfiguration(default=DataRegionConfiguration(persistent=True))
+
+        ignite_config = IgniteConfiguration(
+            version=IgniteVersion(ignite_version),
+            data_storage=data_cfg,
+        )
+
+        service = IgniteService(self.test_context, ignite_config, num_nodes=self.NUM_NODES - 1)
+        service.start()
+
+        control_utility = ControlUtility(service, self.test_context)
+        control_utility.activate()
+
+        client_config = ignite_config._replace(client_mode=True, data_storage=None)
+
+        streamer = IgniteApplicationService(
+            self.test_context,
+            client_config,
+            java_class_name="org.apache.ignite.internal.ducktest.tests.UuidStreamerApplication",
+            params={
+                "cacheName": "test-cache",
+                "iterSize": 512 * 1024
+            }
+        )
+
+        load(streamer, duration=300)
+
+        node = service.nodes[0]
+
+        check_idle_verify(control_utility)
+        check_validate_indexes(control_utility)
+        dump_1 = get_dump_path(control_utility, node)
+
+        self.logger.warn(f'Path to dump_1 on {node.account.externally_routable_ip}={dump_1}')
+
+        data = control_utility.snapshot_create(self.SNAPSHOT_NAME)
+
+        self.logger.debug(data)
+
+        await_snapshot(service, logger=self.logger)
+
+        print_snapshot_size(service, self.SNAPSHOT_NAME, self.logger)
+
+        load(streamer)
+
+        dump_2 = get_dump_path(control_utility, node)
+
+        self.logger.warn(f'Path to dump_2 on {node.account.externally_routable_ip}={dump_2}')
+
+        diff = node.account.ssh_output(f'diff {dump_1} {dump_2}', allow_fail=True)
+        assert len(diff) != 0
+
+        service.stop()
+
+        service.rename_db(new_db_name='old_db')
+        service.copy_snap_to_db(self.SNAPSHOT_NAME)
+
+        service.restart()
+
+        control_utility.activate()
+
+        check_idle_verify(control_utility)
+        check_validate_indexes(control_utility)
+        dump_3 = get_dump_path(control_utility, node)
+
+        self.logger.warn(f'Path to dump_3 on {node.account.externally_routable_ip}={dump_3}')
+
+        diff = node.account.ssh_output(f'diff {dump_1} {dump_3}', allow_fail=True)
+        assert len(diff) == 0, diff
+
+    def copy_service_logs(self, test_status):
```

Comment:
this should be triggered if persistence enabled for any test?

Reply:
No, only if we need it and we explicitly call it in the test

Reply:
Who makes a decision? Should we have a regular flow?

Reply:
the author of the test, by default will be copied only the log